### DISCUSSION TITLE
Optional metadata for transitions

### DISCUSSION THREAD

**Discussion by dwohlfahrt** [Borderless360/django-logic#65] [Ideas]
Would be handy if you could pass in a dict via a metadata kwarg on the Transition constructor to store/link additional information about the given transition.  For my use case, I am exposing the transitions for one of my processes in a web UI where users can manually transition objects from one stage to another.  Being able to associate additional metadata items such as friendly_display_name, description, etc. with each transition would allow me to cleanly incorporate them in the UI vs having to hardcode mappings for friendly names, descriptions, etc.

**Comment by edigerad** [comment]
Thank you for the suggestion! This aligns with the recently added `context` parameter, mentioned in the README, which allows passing a dictionary with custom metadata for transitions. It should fit your use case for associating `friendly_display_name`, `description`, etc. If it doesn't fully meet your needs, feel free to share more details!
